java-util/src/main/java/io/druid/java/util/common/parsers/TimestampParser.java

```diff
@@ -43,7 +43,7 @@ public DateTime apply(String input)
 
           for (int i = 0; i < input.length(); i++) {
             if (input.charAt(i) < '0' || input.charAt(i) > '9') {
-              return new DateTime(ParserUtils.stripQuotes(input));
+              return new DateTime(ParserUtils.replace(ParserUtils.stripQuotes(input), " ", "T"));
```

Comment:
Rather than replacing " " with "T" this could be done more cleanly by making a parser that does what we want. Something like this:

```
  private static DateTimeFormatter createAutoParser()
  {
    final DateTimeFormatter offsetElement = new DateTimeFormatterBuilder()
        .appendTimeZoneOffset("Z", true, 2, 4)
        .toFormatter();

    DateTimeParser timeOrOffset = new DateTimeFormatterBuilder()
        .append(
            null,
            new DateTimeParser[]{
                new DateTimeFormatterBuilder().appendLiteral('T').toParser(),
                new DateTimeFormatterBuilder().appendLiteral(' ').toParser()
            }
        )
        .appendOptional(ISODateTimeFormat.timeElementParser().getParser())
        .appendOptional(offsetElement.getParser())
        .toParser();

    return new DateTimeFormatterBuilder()
        .append(ISODateTimeFormat.dateElementParser())
        .appendOptional(timeOrOffset)
        .toFormatter();
  }
```

And save that in a static and use that instead of `new DateTime(...)`.